app/modules/site_settings/helpers.py

```diff
@@ -276,6 +277,51 @@ def get_definition(cls, cls_name, guid):
                 return defn
         return None
 
+    @classmethod
+    # replace=False will silently fail if already exists
+    def add_definition(cls, class_name, guid, defn, replace=False):
```

Comment:
OPther than the tests, this is not  called from anywhere. What's it actually here for?

Reply:
i had originally planned to use it but then only really needed `remove_definition()`.  so i just left it around.   it can be dropped as i dont think there is going to be any way any of the future tickets will need this.